pkg/controller/tidbcluster_control.go

```diff
@@ -78,7 +80,9 @@ func (rtc *realTidbClusterControl) UpdateTidbCluster(tc *v1alpha1.TidbCluster) (
 
 		return updateErr
 	})
-	rtc.recordTidbClusterEvent("update", tc, err)
+	if !deepEqualExceptHeartbeatTime(new, old) {
```

Comment:
`new` and `old` are pointers, this method actually modifies the status. Even it's not used anymore, you should avoid modification.